src/lib/utilities/conversion/units.ts

```diff
@@ -342,3 +344,96 @@ function makeUnit(unit: UnitOptions): Unit {
 			: { from: (si) => div(si, unit.value), to: (value) => mul(value, unit.value) }
 	};
 }
+
+export function sanitizeUnit(unit: string) {
+	return unit.replaceAll(/(º)|\^(\d+)/g, (_, degree, number) => (degree ? '°' : toSuperscript(number)));
+}
+
+export function renderUnit(t: TFunction, unit: Unit) {
+	let name = t(unit.name);
+	if (unit.prefixMultiplier) name = t(LanguageKeys.Units.PrefixUnit, { prefix: t(unit.prefixMultiplier), unit: name });
+	if (unit.prefixDimension) name = t(LanguageKeys.Units.PrefixDimension, { dimension: t(unit.prefixDimension), unit: name });
+	return name;
+}
+
+export const defaults = [
+	'm', // Meter
+	'cm', // Centimeter
+	'km', // Kilometer
+	'mm', // Millimeter
+	'dm', // Decimeter
+	'dam', // Decameter
+	'lb', // Pound
+	'kg', // Kilogram
+	'US gal', // Gallon
+	'L', // Liter
+	'ft', // Foot
+	'mi', // Mile
+	'fl oz', // Fluid Ounce
+	'in' // Inch
+].map((u) => Units.get(u)!);
+
+const NameMappings = new Collection<LocaleString, Collection<string, Set<string>>>();
+
+function createNameMappings(locale: LocaleString) {
+	const collection = new Collection<string, Set<string>>();
+	const t = getT(locale);
+	for (const unit of Units.values()) {
+		const name = renderUnit(t, unit);
+		collection.ensure(unit.symbol, () => new Set()).add(name.toLowerCase());
+	}
+	return collection;
+}
+
+const enUSNameMappings = createNameMappings('en-US');
+NameMappings.set('en-US', enUSNameMappings);
+for (const locale of loadedLocales) {
+	if (locale === 'en-US') continue;
+	NameMappings.set(locale, createNameMappings(locale));
+	const collection = NameMappings.get(locale)!;
+	for (const [symbol, names] of enUSNameMappings.entries()) {
+		collection.get(symbol)!.add([...names][0]);
+	}
+}
+
+export function getNameMappings(locale: LocaleString) {
+	return NameMappings.get(locale) ?? NameMappings.get('en-US')!;
+}
+
+const MaximumLength = 100;
+
+export function searchUnits(id: string, locale: LocaleString): readonly UnitSearchResult[] {
+	if (id.length === 0) return defaults.map((value) => ({ score: 1, value }));
+	if (id.length > MaximumLength) return [];
+
+	id = sanitizeUnit(id).toLowerCase();
+	const entries = [] as UnitSearchResult[];
+	for (const [symbol, names] of getNameMappings(locale).entries()) {
+		const unit = Units.get(symbol)!;
+
+		let highestScoredResult = 0;
+		for (const name of names) {
+			const score = getSearchScore(id, symbol, name);
+			if (score > highestScoredResult) {
+				highestScoredResult = score;
+			}
+		}
+		if (highestScoredResult !== 0) entries.push({ score: highestScoredResult, value: unit });
```

Comment:
You can simplify the code further by using `Math.max`:
```suggestion
		let score = 0;
		for (const name of names) {
			score = Math.max(score, getSearchScore(id, symbol, name));
		}
		if (score !== 0) entries.push({ score, value: unit });
```
Also renamed the variable to `score` to reduce noise in the last line.